kyaml/fn/runtime/runtimeutil/functiontypes.go

```diff
@@ -204,7 +204,7 @@ func (s *StorageMount) String() string {
 func GetFunctionSpec(n *yaml.RNode) (*FunctionSpec, error) {
 	meta, err := n.GetMeta()
 	if err != nil {
-		return nil, nil
+		return nil, fmt.Errorf("failed to get ResourceMeta: %w", err)
```

Comment:
Although these may be good changes, they look unrelated to this PR. Is a new linter somehow triggering the need for them?

Reply:
It was caused by the `nilerr` with the new version.
That is not a new linter, that linter was enabled before creating this PR.

That lint shows the below errors.
```
error is not nil (line 205) but it returns nil (nilerr)
```

Reply:
Ah, I see. This was probably done to make the function alpha as unlikely as possible to generate errors, but now that we're trying to move functions forward, I think you're right to return the error here. It would be very surprising to get an error here, and if somehow that happened, we don't know whether or not the metadata we failed to retrieve contained a function.